signer/signer.go

```diff
@@ -36,24 +36,34 @@ import (
 // signingAgent is the unprotected header field used by signature.
 const signingAgent = "Notation/1.0.0"
 
-// genericSigner implements notation.Signer and embeds signature.Signer
-type genericSigner struct {
+// GenericSigner implements notation.Signer and embeds signature.Signer
+type GenericSigner struct {
```

Comment:
What is the reason for making this public?

Reply:
GenericSigner satisfies both blob signer and container signer interface so instead of returning notatin.Signer now we need to return `GenericSigner`